causalml/inference/meta/drlearner.py

```diff
@@ -0,0 +1,695 @@
+            tau (np.array): a treatment effect vector (estimated/actual)
+            model_tau_feature (sklearn/lightgbm/xgboost model object): an unfitted model object
+            features (optional, np.array): list/array of feature names. If None, an enumerated list will be used.
+            shap_dict (optional, dict): a dict of shapley value matrices. If None, shap_dict will be computed.
+            interaction_idx (optional, str or int): feature index / name used in coloring scheme as interaction feature.
+                If "auto" then shap.common.approximate_interactions is used to pick what seems to be the
+                strongest interaction (note that to find to true strongest interaction you need to compute
+                the SHAP interaction values).
+        """
+        override_checks = False if shap_dict is None else True
+        explainer = Explainer(
+            method="shapley",
+            control_name=self.control_name,
+            X=X,
+            tau=tau,
+            model_tau=model_tau_feature,
+            features=features,
+            override_checks=override_checks,
+            classes=self._classes,
+        )
+        explainer.plot_shap_dependence(
+            treatment_group=treatment_group,
+            feature_idx=feature_idx,
+            shap_dict=shap_dict,
+            interaction_idx=interaction_idx,
+            **kwargs
+        )
+
+class BaseDRRegressor(BaseDRLearner):
+    """
+    A parent class for DR-learner regressor classes.
+    """
+
+    def __init__(self,
+                 learner=None,
+                 control_outcome_learner=None,
+                 treatment_outcome_learner=None,
+                 treatment_effect_learner=None,
+                 ate_alpha=.05,
+                 control_name=0):
+        """Initialize an DR-learner regressor.
+
+        Args:
+            learner (optional): a model to estimate outcomes and treatment effects in both the control and treatment
+                groups
+            control_outcome_learner (optional): a model to estimate outcomes in the control group
+            treatment_outcome_learner (optional): a model to estimate outcomes in the treatment group
+            control_effect_learner (optional): a model to estimate treatment effects in the control group
+            treatment_effect_learner (optional): a model to estimate treatment effects in the treatment group
+            ate_alpha (float, optional): the confidence level alpha of the ATE estimate
+            control_name (str or int, optional): name of control group
+        """
+        super().__init__(
+            learner=learner,
+            control_outcome_learner=control_outcome_learner,
+            treatment_outcome_learner=treatment_outcome_learner,
+            treatment_effect_learner=treatment_effect_learner,
+            ate_alpha=ate_alpha,
+            control_name=control_name)
+
+class XGBTDRRegressor(BaseDRRegressor):
```

Comment:
To be consistent with `XGBTRegressor` for XGBoost T-regressor, let's use `XGBDRRegressor` without `T` here.

Reply:
Done!